code/Weapons/Gadget/Gadget.cs

```diff
@@ -125,4 +127,12 @@ public void PlayScreenSound( string sound )
 	{
 		this.SoundFromScreen( sound );
 	}
+
+	private IEnumerable<GadgetComponent> GetSortedComponents()
+	{
+		if ( _sortCache is null )

```

Comment:
The only issue with this is if new components get added or if the sort order changes, then it will not update the sort order.

@trundlr are we concerned with a performance hit if we just sort on every single tick?

Reply:
Nope, these gadgets only have a couple components to begin with, the sort is probably nlogn, meaning we're likely seeing negligible overhead on each gadget.

That said, how often do we change the sort order or add/remove components after spawning? I assume if it was ever an issue we could simply re-sort the component order on component add/remove or sort order change manually.

Reply:
Currently, SortOrder is never changed as it's brand new, and as far as I can tell, no components are added/removed after a weapon is spawned, either.